src/main/java/com/cloudbees/hudson/plugins/folder/views/AbstractFolderViewHolder.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2016 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.cloudbees.hudson.plugins.folder.views;
+
+import com.cloudbees.hudson.plugins.folder.AbstractFolder;
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.ExtensionPoint;
+import hudson.model.View;
+import hudson.views.ViewsTabBar;
+import java.util.List;
+
+/**
+ * Responsible for holding the view configuration of an {@link AbstractFolder}. Each {@link AbstractFolder} concrete
+ * type should define its view configuration holder by returning the implementaion from
+ * {@link AbstractFolder#newFolderViewHolder()}
+ *
+ * Use-cases:
+ * <ul>
+ * <li>
+ * Where the {@link AbstractFolder} permits the views to be configured by the user, use a
+ * {@link DefaultFolderViewHolder}
+ * </li>
+ * <li>
+ * Where the {@link AbstractFolder} has a fixed set of pre-configured views, the plugin can provide
+ * a custom implementation that returns the fixed set of views.
+ * </li>
+ * </ul>
+ *
+ * @since FIXME
+ */
+public abstract class AbstractFolderViewHolder implements ExtensionPoint {
```

Comment:
Why is this an `ExtensionPoint`? IIUC it comes from `newFolderViewHolder`, not an `@Extension`.